### ISSUE TITLE
Strange behavior with Alt + Tab and window_decorations = "NONE"

### ISSUE THREAD

**Issue by gmr458** [wez/wezterm#2375]
### What Operating System(s) are you seeing this problem on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

Mutter

### WezTerm version

20220808-084502-d5755f3e

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

When i maximize the wezterm window and switch between windows with Alt + Tab the wezterm window sticks to the top left of the screen, also when only wezterm is open and the window is maximized and i press the windows key on my keyboard to see all windows in GNOME the wezterm window takes up too much space. All this only happens when `window_decorations = "NONE"`.
<video src="https://user-images.githubusercontent.com/53343401/183479260-48e48e0b-cbc7-40e9-8e2c-7ac3f97ea303.mp4" width="100%"></video>

### To Reproduce

1. Open wezterm and maximize it.
2. Open another program.
3. Switch to the wezterm window with Alt + Tab

### Configuration

```lua
local wezterm = require("wezterm")

return {
	font = wezterm.font("Iosevka Fixed", { weight = "Regular", italic = false, bold = false }),
	font_rules = {
		{
			intensity = "Bold",
			font = wezterm.font("Iosevka Fixed", { weight = "Regular" }),
		},
	},
	font_size = 12.0,
	enable_tab_bar = false,
	window_decorations = "NONE",
}
```

### Expected Behavior

switch between windows with Alt + Tab and the wezterm window does not change its size and position

**Comment by nogden** [comment]
I also have this bug with Wezterm and was having the exact same issue with Kitty under the same circumstances. [This commit](https://github.com/kovidgoyal/kitty/pull/4811/commits/99a5843595779bbaa51355fc97fbd8afd8e89737) fixed the issue in Kitty. So I thought I'd leave this here just in case it turns out to be the same interaction with Wayland that's causing the issue for Wezterm.

Apologies in advance if it turns out not to be useful.

**Comment by aliaksandr-trush** [comment]
Should be fixed by #5897
